Sources/GenericModule/Presenter/Presenter.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Copyright © 2021 Rosberry. All rights reserved.
+//
+
+open class Presenter<View: GenericModule.View,
+                     Input,
+                     Output,
+                     Dependencies>: ModulePresenter, ViewOutput {
+
+    public typealias ViewModel = View.ViewModel
+    public typealias State = ViewModel.State
+
+    public var state: State
+    public var output: Output?
+
+    weak var view: View?
```

Comment:
I wonder if we can make it as unowned and view become not optional

Reply:
view input is not optional now